File: Lista01/PY/ex09.py
from abc import ABC, abstractmethod

# Definindo a classe abstrata Imprimivel
class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self) -> None:
        pass

# Implementando a classe Relatorio
class Relatorio(Imprimivel):
    def imprimir(self) -> None:
        print("Imprimindo relatório...")

# Implementando a classe Contrato
class Contrato(Imprimivel):
    def imprimir(self) -> None:
        print("Imprimindo contrato...")

# Função para demonstrar a impressão
def imprimir_documento(documento: Imprimivel) -> None:
    documento.imprimir()

# Exemplo de uso
relatorio = Relatorio()
contrato = Contrato()

imprimir_documento(relatorio)
imprimir_documento(contrato)

File: Lista01/GO/ex09.go
package main

import "fmt"

// Definindo a interface Imprimivel
type Imprimivel interface {
    Imprimir()
}

// Implementando a interface em Relatorio
type Relatorio struct{}

func (r Relatorio) Imprimir() {
    fmt.Println("Imprimindo relatório...")
}

// Implementando a interface em Contrato
type Contrato struct{}

func (c Contrato) Imprimir() {
    fmt.Println("Imprimindo contrato...")
}

// Função principal para demonstrar a impressão
func main() {
    var relatorio Imprimivel = Relatorio{}
    var contrato Imprimivel = Contrato{}

    relatorio.Imprimir()
    contrato.Imprimir()
}
